[chart/banzaicloud-stable/thanos] Helm 3 template validation errors on rule installation

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
When installing the thanos chart with Helm 3 the following validation errors fail the installation:

coalesce.go:196: warning: cannot overwrite table with non table for annotations (map[])
coalesce.go:165: warning: skipped value for objstore: Not a table.
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(StatefulSet.spec.template.spec): unknown field "resources" in io.k8s.api.core.v1.PodSpec, ValidationError(StatefulSet.spec.template.spec): unknown field "secretName" in io.k8s.api.core.v1.PodSpec, ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec]

**Steps to reproduce the issue:**
1) Enable rule component.
2) Install chart using Helm 3 with the following command: helm install -f values.yaml thanos banzaicloud-stable/thanos

**Expected behavior**
Successful installation of rule statefulset.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Additional context**
If rule component is disabled then the rest of the chart components install successfully.

--- a/thanos/Chart.yaml
+++ b/thanos/Chart.yaml
@@ -9,7 +9,7 @@ keywords:
 sources:
   - https://github.com/thanos-io/thanos
   - https://github.com/banzaicloud/banzai-charts/tree/master/thanos
-version: 0.3.12
+version: 0.3.13
 icon: https://raw.githubusercontent.com/thanos-io/thanos/master/docs/img/Thanos-logo_fullmedium.png
 maintainers:
 - name: Banzai Cloud

--- a/thanos/templates/rule-statefulset.yaml
+++ b/thanos/templates/rule-statefulset.yaml
@@ -23,6 +23,7 @@ spec:
       app.kubernetes.io/name: {{ include "thanos.name" . }}
       app.kubernetes.io/instance: {{ .Release.Name }}
       app.kubernetes.io/component: rule
+  serviceName: {{ include "thanos.name" . }}
   template:
     metadata:
       labels:
@@ -34,6 +35,7 @@ spec:
       - name: thanos-rule
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        resources: {{ toYaml .Values.rule.resources | nindent 10 }}
         {{- with .Values.rule.extraEnv }}
         env: {{ toYaml . | nindent 8 }}
         {{- end }}
@@ -115,14 +117,12 @@ spec:
       - name: rule-volume
         configMap:
           name: {{ include "thanos.fullname" . }}-rules
-      secretName: {{ include "thanos.fullname" . }}
       {{- if .Values.rule.livenessProbe }}
       livenessProbe: {{ toYaml .Values.rule.livenessProbe | nindent 8 }}
       {{- end }}
       {{- if .Values.rule.readinessProbe }}
       readinessProbe: {{ toYaml .Values.rule.readinessProbe | nindent 8 }}
       {{- end }}
-      resources: {{ toYaml .Values.rule.resources | nindent 8 }}
       {{- with .Values.rule.securityContext }}
       securityContext: {{ toYaml . | nindent 8 }}
       {{- end }}

